packages/rum/test/readReplayPayload.ts

```diff
@@ -0,0 +1,60 @@
+import pako from 'pako'
+
+import type { Payload } from '@datadog/browser-core'
+import type { BrowserSegment, CreationReason } from '../src/types'
+import type { BrowserSegmentMetadataAndSegmentSizes } from '../src/domain/segmentCollection'
+
+export async function readReplayPayload(payload: Payload) {
+  return {
+    segment: await readSegmentFromReplayPayload(payload),
+    metadata: await readMetadataFromReplayPayload(payload),
+  }
+}
+
+function readSegmentFromReplayPayload(payload: Payload) {
+  return readJsonBlob((payload.data as FormData).get('segment') as Blob, {
+    decompress: true,
+  }) as Promise<BrowserSegment>
+}
+
+export function readMetadataFromReplayPayload(payload: Payload) {
+  const formData = payload.data as FormData
+  if (!formData.has('event')) {
+    // TODO remove this when replay_json_payload is enabled
+    return {
+      application: { id: formData.get('application.id') as string },
+      session: { id: formData.get('session.id') as string },
+      view: { id: formData.get('view.id') as string },
+      start: Number(formData.get('start')),
+      end: Number(formData.get('end')),
+      records_count: Number(formData.get('records_count')),
+      source: formData.get('source') as 'browser',
+      creation_reason: formData.get('creation_reason') as CreationReason,
+      raw_segment_size: Number(formData.get('raw_segment_size')),
+      index_in_view: Number(formData.get('index_in_view')),
+      has_full_snapshot: formData.get('has_full_snapshot') === 'true',
+    } satisfies Omit<BrowserSegmentMetadataAndSegmentSizes, 'compressed_segment_size'> & {
+      compressed_segment_size?: number
+    }
+  }
+
+  return readJsonBlob((payload.data as FormData).get('event') as Blob) as Promise<BrowserSegmentMetadataAndSegmentSizes>
```

Comment:
**question:** why not cast in the `resolve` function in `readJsonBlob`?

Reply:
You mean, something like having something like this?

```ts
function readJsonBlob<T>() {
  return new Promise<T>(...)
}
```

The end result is indeed the same, but the cast is implicit. I rather have a `readJsonBlob(): Promise<unknown>`, which provide no guarantee about the returned type, and then let the caller take the responsibility to ducktype it or cast it. But I agree the syntax is a bit more verbose.

Reply:
this would not work?
```ts
resolve(deserialized as BrowserSegmentMetadataAndSegmentSizes)
```

Reply:
`readJsonBlob()` is used two times with different return types